src/Sarif.WorkItems/SarifWorkItemFiler.cs

```diff
@@ -232,6 +232,11 @@ public virtual IReadOnlyList<SarifLog> SplitLogFile(SarifLog sarifLog)
                             partitionFunction = (result) => result.ShouldBeFiled() ? result.GetPerRepositoryFingerprintSplittingStrategyId() : null;
                             break;
                         }
+                        case SplittingStrategy.PerFingerprint:
+                        {
+                            partitionFunction = (result) => result.ShouldBeFiled() ? result.Fingerprints.Count == 0 ? null : result.Fingerprints.First().Value : null;
```

Comment:
>First() [](start = 145, length = 7)

I'm fixating on "First" here.  In this splitting strategy, if there are multiple results with the same fingerprint, the rest after the first will be discarded, right?  Is that desired?

Reply:
i didn't want to hard code a specific key...its why i added first. i could also add a parameter being the fingerprint name. 

---
In reply to: [581473453](https://github.com/microsoft/sarif-sdk/pull/2293#discussion_r581473453) [](ancestors = 581473453)

Reply:
I have no strong preference so long as we're preserving all the intended information :)

---
In reply to: [581474839](https://github.com/microsoft/sarif-sdk/pull/2293#discussion_r581474839) [](ancestors = 581474839,581473453)